### PR TITLE
Smart difference detection

### PR DESCRIPTION
**Pull request by Quramy**
Works for #90 . 

What I did:

> Invoke new engine from reg-cli/report. (OnDemand detection)

report.html can communicate x-img-diff-js's Web Workers. You can check the operation `npm run dev:report` and click the actual( or expected) image in the "Changed Items" section 😄 

[https://gyazo.com/493043a31e55b3de63777939a797a494](https://gyazo.com/493043a31e55b3de63777939a797a494)

> Introduce CLI option to tell reg-cli "enable smart diff"

User can enable `x-img-diff` via `-X client` option.

> Add new view components to review the new comparison result

I've created `ComparisonModal` to draw diff-marker rectangular over 2 images(left: actual, right: expected).

And I want to make more interactive them. For example:

- When user hover on diff a rectangle in actual view, indicates the corresponding rectangle in expected view
- When user mouseover a region part in actual/expected, indicate translation information, etc...

However, modal backdrop obstacles mouse interaction, and these features are not essential. So  I'll create it later 🙏

### CODE REVIEW

**Comment by bokuweb** [comment]
When I opened modal, below error occurred.



## step

- clone this branch
- yarn install
- npm run dev:report
- click image

## browser

- バージョン: 62.0.3202.94
- バージョン: 64.0.3270.0

File: report/src/worker-client.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        case 'res_calc':
+          this.pushResult(ev.data);
+          break;
+        default:
+      }
+    });
+  }
+}
+
+const clinet = new WorkerClient();
```
Comment (bokuweb) [inline report/src/worker-client.js]: [May] client 🚓

File: src/cli.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ const cli = meow(`
     -T, --threshold Threshold for detecting change. Value can range from 0.00 (no difference) to 1.00 (every pixel is different).
     -C, --concurrency How many processes launches in parallel. If omitted 4.
     -A, --enableAntialias. Enable antialias. If omitted false.
+    -X, --additionalDetection. Enable additional deffirence detection(highly experimental). Select "none" or "client" (default: "none").
```
Comment (bokuweb) [inline src/cli.js]: [May] `difference`

File: test/cli.test.js [hunk 3]
```diff
@@ -143,6 +143,28 @@ test.serial('should generate report html to `${WORKSPACE}/dist/report.html` when
   }
 });
 
+test.serial('should generate worker js and wasm files under `${WORKSPACE}/dist` when `-R -X` option enabled', async t => {
```
Comment (bokuweb) [inline test/cli.test.js]: 👍

File: report/src/views/ComparisonModal.vue [hunk 4]
```diff
@@ -0,0 +1,132 @@
+<template>
+  <div>
+    <modal name="comparison">
+      <div class="wrapper">
+        <div class="modal">
+          <div class="markers-area">
+            <img :src="srcActual" />
+            <div v-if="matching">
+              <div class="markers" v-for="m in matching.matches">
```
[the remark this answers is not in the document]
Comment (Quramy) [reply report/src/views/ComparisonModal.vue]: I got it. However, the item of `matches` does not have property which can be treated as `:key`. Structure of this item: 

```js
[
  {  /* the first item is corresponding to actual image */
    boundingRect: { ... }, /* bounding box of this iterator */
    diffMarkers: [...], /* array of rect surrounding difference points */
    center: { ... }, /* center of the bounding box */
  },
  {  /* the second item is corresponding to expected image */
    boundingRect: { ... }, /* bounding box of this iterator */
    diffMarkers: [...], /* array of rect surrounding difference points */
    center: { ... }, /* center of the bounding box */
  }
]
```

[the remark this answers is not in the document]
Comment (bokuweb) [reply report/src/views/ComparisonModal.vue]: I see. Let's leave it as it is.

File: report/src/detector-wrapper/util.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
+// 1. +++ fetchAndInstantiate() +++ //
```
[the remark this answers is not in the document]
Comment (Quramy) [reply report/src/detector-wrapper/util.js]: Yes. I copied and modify a little(added some exception handling).

**Comment by Quramy** [comment]
🎉  🎉

**Pull request merged**